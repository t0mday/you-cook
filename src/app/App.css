.App {
  text-align: center;
}

.App-header {
  background-image: url('../images/creveta-1327214.jpg');
  background-repeat: no-repeat;
  background-size: cover;
  background-position: 60% 15%;
  height: 600px;
  display: flex;
  justify-content: center;
  align-items: center;
}

.banner {
  font-size: 4rem;
  background-color: rgba(250, 235, 215, 0.8);
  margin: 0;
  width: 100%;
  padding: 2rem;
}

h1 {
  color: rgb(83,153,0);
}

h1 span {
  font-style: italic;
}

footer p {
  text-align: right;
  font-style: italic;
  padding: 30px;
  font-family: Helvetica, Tahoma, sans-serif;
}

footer img {
  float: left;
  width: 200px;
  border: none;
}

section {
  border-bottom: 3px solid rgb(17, 54, 112);
}
