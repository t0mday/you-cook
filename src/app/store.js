import { configureStore } from '@reduxjs/toolkit';
import recipesReducer from '../components/recipes/recipesSlice';
import searchReducer from '../components/search/searchSlice';


export const store = configureStore({
  reducer: {
    recipes: recipesReducer,
    search: searchReducer
  }
});
