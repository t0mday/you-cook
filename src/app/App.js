import React from 'react';
import './App.css';
import attribution from './Edamam_Badge_Transparent.svg';
import Search from '../components/search/Search';
import SearchResults from '../components/searchResults/SearchResults';

function App() {

  return (
    <div className="App">
      <header className="App-header">
        <h1>you Cook</h1>
      </header>
      <section className='favourites'>
        <Search />
      </section>
      <section className='search'>
        {/*<Favourites />*/}
      </section>
      <section className='searchResults'>
        <SearchResults />
      </section>
      <footer>
        <img src={attribution} alt="powered by Edamam" />
      </footer>
    </div>
  );
}

export default App;
