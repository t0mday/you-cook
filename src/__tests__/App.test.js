import React from 'react';
import { render } from '@testing-library/react';
import App from '../app/App';
import { store } from '../../src/app/store';
import { Provider } from 'react-redux';


test('renders correctly', () => {
  const { asFragment } = render(<Provider store={store}><App /></Provider>);
  expect(asFragment()).toMatchSnapshot();
});