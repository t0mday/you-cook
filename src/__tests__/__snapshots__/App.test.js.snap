// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <div
        class="banner"
      >
        <h1>
          You Cook
        </h1>
      </div>
    </header>
    <section
      class="search"
    >
      <h2>
        Find Recipes
      </h2>
      <form>
        <input
          id="searchField"
          name="searchField"
          placeholder="Enter food to search"
          required=""
          type="text"
          value=""
        />
        <button
          type="submit"
        >
          Search Recipes
        </button>
        <p
          class="noShowPrompt"
        >
          Scroll down to see results
        </p>
        <div
          class="checkboxes"
        >
          <fieldset
            class="restrictions"
          >
            <legend>
              Restrictions
            </legend>
            <label
              for="gluten-free"
            >
              <input
                id="gluten-free"
                type="checkbox"
              />
              Gluten-free
            </label>
            <label
              for="dairy-free"
            >
              <input
                id="dairy-free"
                type="checkbox"
              />
              Dairy-free
            </label>
            <label
              for="vegetarian"
            >
              <input
                id="vegetarian"
                type="checkbox"
              />
              Vegetarian
            </label>
            <label
              for="vegan"
            >
              <input
                id="vegan"
                type="checkbox"
              />
              Vegan
            </label>
            <label
              for="egg-free"
            >
              <input
                id="egg-free"
                type="checkbox"
              />
              Egg-free
            </label>
            <label
              for="peanut-free"
            >
              <input
                id="peanut-free"
                type="checkbox"
              />
              Peanut-free
            </label>
            <label
              for="tree-nut-free"
            >
              <input
                id="tree-nut-free"
                type="checkbox"
              />
              Tree-nut-free
            </label>
            <label
              for="shellfish-free"
            >
              <input
                id="shellfish-free"
                type="checkbox"
              />
              Shellfish-free
            </label>
            <label
              for="pork-free"
            >
              <input
                id="pork-free"
                type="checkbox"
              />
              Pork-free
            </label>
            <label
              for="kosher"
            >
              <input
                id="kosher"
                type="checkbox"
              />
              Kosher
            </label>
          </fieldset>
          <fieldset
            class="cuisine"
          >
            <legend>
              Cuisine
            </legend>
            <label
              for="Asian"
            >
              <input
                id="Asian"
                type="checkbox"
              />
              Asian
            </label>
            <label
              for="Caribbean"
            >
              <input
                id="Caribbean"
                type="checkbox"
              />
              Caribbean
            </label>
            <label
              for="Chinese"
            >
              <input
                id="Chinese"
                type="checkbox"
              />
              Chinese
            </label>
            <label
              for="Eastern Europe"
            >
              <input
                id="Eastern Europe"
                type="checkbox"
              />
              East Europe
            </label>
            <label
              for="French"
            >
              <input
                id="French"
                type="checkbox"
              />
              French
            </label>
            <label
              for="Indian"
            >
              <input
                id="Indian"
                type="checkbox"
              />
              Indian
            </label>
            <label
              for="Italian"
            >
              <input
                id="Italian"
                type="checkbox"
              />
              Italian
            </label>
            <label
              for="Japanese"
            >
              <input
                id="Japanese"
                type="checkbox"
              />
              Japanese
            </label>
            <label
              for="Mediterranean"
            >
              <input
                id="Mediterranean"
                type="checkbox"
              />
              Mediterranean
            </label>
            <label
              for="Mexican"
            >
              <input
                id="Mexican"
                type="checkbox"
              />
              Mexican
            </label>
          </fieldset>
        </div>
      </form>
    </section>
    <hr />
    <div
      class="all-recipes"
    >
      <section
        class="favourites"
      >
        <h2>
          Favourites
        </h2>
        <div
          class="recipes"
        />
      </section>
      <hr />
      <section
        class="searchResults"
      >
        <h2>
          Search Results
        </h2>
        <div
          class="recipes"
        />
      </section>
    </div>
    <footer>
      <img
        alt="powered by Edamam"
        src="Edamam_Badge_Transparent.svg"
      />
      <p>
        Â© Tom Day 2022
      </p>
    </footer>
  </div>
</DocumentFragment>
`;
